ci/before_cache.sh

```diff
@@ -19,8 +15,14 @@ ccache -s 2>/dev/null || true
 
 # Update the third-party dependency cache only if the build was successful.
 if ended_successfully; then
-  rm -rf "${HOME}/.cache/nvim-deps"
-  mv "${DEPS_BUILD_DIR}" "${HOME}/.cache/nvim-deps"
+  if ! test -f "${CACHE_MARKER}" ; then
+    # Fresh cache, remove cruft from Travis base image.
+    rm -rf "${HOME}/.cache"
+    mkdir "${HOME}/.cache"
+  fi
+
+  test -e "${CACHE_NVIM_DEPS}" && rm -rf "${CACHE_NVIM_DEPS}"
```

Comment:
why `test -e` before `rm -rf` ?

Reply:
Extra-safety.

Likely safe due to being quoted (?), but CACHE_NVIM_DEPS could be empty (e.g. when running this by itself manually etc).

Reply:
yeah, all of these do nothing on bash 3.2 (macOS):

```
rm -rf "$FOO"
rm -rf ""
rm -rf
```